django_mongodb/compiler.py

```diff
@@ -0,0 +1,348 @@
+from functools import cached_property
+from itertools import chain
+
+from django.db.models.sql.compiler import (
+    SQLCompiler as BaseSQLCompiler,
+)
+from django.db.models.sql.compiler import (
+    SQLInsertCompiler as BaseSQLInsertCompiler,
+)
+from django.db.models.sql.constants import (
+    CURSOR,
+    GET_ITERATOR_CHUNK_SIZE,
+    MULTI,
+    NO_RESULTS,
+    SINGLE,
+)
+from pymongo import InsertOne, UpdateOne
+
+from django_mongodb.query import MongoOrdering, MongoSelect, MongoWhereNode
+
+
+class SQLCompiler(BaseSQLCompiler):
+    def __init__(self, query, connection, using, elide_empty=True):
+        super().__init__(query, connection, using, elide_empty)
+        self.extr = None
+
+    def build_mongo_filter(self, filter_expr):
+        referenced_tables = set()
```

Comment:
A bare predicate filter, which is used as `$match` in query pipelines or as filter in update/delete.